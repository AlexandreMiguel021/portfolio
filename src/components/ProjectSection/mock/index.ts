export const allProjects = [
	{
		id: 1,
		title: 'vrstore',
		description: 'vrstore é um site...',
		tech: '<ul>\n  <li>HTML</li>\n  <li>CSS</li>\n  <li>JavaScript</li>\n</ul>',
		repository: 'https://github.com',
		livesite: 'https://vr-store-mu.vercel.app'
	},
	{
		id: 2,
		title: 'portfolio',
		description: 'portfolio é um site...',
		tech: '<ul>\n  <li>HTML</li>\n  <li>CSS</li>\n  <li>JavaScript</li>\n</ul>',
		repository: 'https://github.com',
		livesite: 'https://portfolio-mu.vercel.app'
	}
]
