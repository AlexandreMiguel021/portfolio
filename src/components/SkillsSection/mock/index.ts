export const allSkills = [
	{
		id: 8882604,
		title: 'HTML',
		icon: {
			url: '/html.svg',
			alt: 'Logotipo HTML'
		}
	},
	{
		id: 8882605,
		title: 'CSS',
		icon: {
			url: '/css.svg',
			alt: 'Logotipo CSS'
		}
	},
	{
		id: 8882606,
		title: 'ReactJS',
		icon: {
			url: '/reactJS.svg',
			alt: 'Logotipo ReactJS'
		}
	}
]
