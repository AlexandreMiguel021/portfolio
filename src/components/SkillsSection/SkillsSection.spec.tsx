import { render, screen } from '@testing-library/react'
import { SkillsSection } from '.'
import { allSkills } from './mock'
import '@testing-library/jest-dom'

describe('SkillsSection', () => {
	it('should render a SkillsSection correctly', () => {
		const { container } = render(<SkillsSection allSkills={allSkills} />)
		expect(container).toMatchSnapshot()
	})

	it('should render a skills section with alt text', () => {
		render(<SkillsSection allSkills={allSkills} />)
		expect(screen.getByAltText('Logotipo HTML')).toBeInTheDocument()
	})

	it('should render a skills section with title card', () => {
		render(<SkillsSection allSkills={allSkills} />)
		expect(screen.getByText('HTML')).toBeInTheDocument()
	})

	it('should render a skill section with three cards', () => {
		render(<SkillsSection allSkills={allSkills} />)
		expect(screen.getByTestId('list').children).toHaveLength(3)
	})
})
