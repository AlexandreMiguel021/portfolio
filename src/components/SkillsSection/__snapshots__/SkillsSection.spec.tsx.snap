// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsSection should render a SkillsSection correctly 1`] = `
<div>
  <section
    id="skills"
  >
    <div
      class="wrapper"
      data-aos="fade-up"
    >
      <h2
        class="heading"
        data-aos="fade-up"
      >
        Habilidades
      </h2>
      <p
        class="description"
      >
        Estas s√£o algumas das ferramentas que estou usando atualmente:
      </p>
      <ul
        class="wrapperCard"
        data-testid="list"
      >
        <li
          class="card"
          data-aos="flip-left"
        >
          <div
            class="img"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Logotipo HTML"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="title"
          >
            HTML
          </p>
        </li>
        <li
          class="card"
          data-aos="flip-left"
        >
          <div
            class="img"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Logotipo CSS"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="title"
          >
            CSS
          </p>
        </li>
        <li
          class="card"
          data-aos="flip-left"
        >
          <div
            class="img"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Logotipo ReactJS"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <p
            class="title"
          >
            ReactJS
          </p>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
